Action.h

```diff
@@ -154,7 +154,7 @@ namespace gameController{
         auto executeAll() const ->
             std::vector<std::pair<gameModel::Environment, double>> override;
 
-    private:
+    //private:
```

Comment:
???

Reply:
Will die Methode testen, kam aber nicht dran, wenn sie private ist.

Reply:
Dann von außen irgendwie testen. So kompilieren die Tests sofort nicht mehr sobald alles die richtige Visibility hat...

Reply:
Von außen testen ist an der stelle schwierig. Die einzige funktion die das nutzt und die man von außen aufrufen kann lässt leider nur noch eingeschränkt rückschlüsse darauf zu ob die eigentlich funktion richtig arbeitet. Also gibt es da in meinen Augen nur die Möglichkeit entweder man kann die Funktion testen oder sie ist private und praktisch nicht zu testen

Reply:
Wenn du sie nicht sichtbar machen willst kannst du sie auch protected machen, dann draus vererben und eine Hilfsfunktion basteln:

Im richtigen Code:
```C++
class A {
...
private:
    int a();
}
```

Irgendwo im Test Code:
```C++
class ATest : public A {
public:
    int aTest() {
        return this->a();
    }
}
```